<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FastModelAlign</class>
 <widget class="qMRMLWidget" name="FastModelAlign">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>778</width>
    <height>800</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabsWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="alignSingleTab">
      <attribute name="title">
       <string>Single Alignment</string>
      </attribute>
      <layout class="QVBoxLayout" name="alignSingleLayout">
       <item>
        <widget class="ctkCollapsibleButton" name="alignSingleWidget">
         <property name="text">
          <string>Set up source and target models</string>
         </property>
         <property name="collapsed">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="FormLayout_1">
          <item row="0" column="0">
           <widget class="QLabel" name="sourceModelLabel">
            <property name="text">
             <string>Source Model: </string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="qMRMLNodeComboBox" name="sourceModelSelector">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="toolTip">
             <string>Select source model</string>
            </property>
            <property name="nodeTypes">
             <stringlist>
              <string>vtkMRMLModelNode</string>
             </stringlist>
            </property>
            <property name="noneEnabled">
             <bool>true</bool>
            </property>
            <property name="addEnabled">
             <bool>false</bool>
            </property>
            <property name="removeEnabled">
             <bool>true</bool>
            </property>
            <property name="editEnabled">
             <bool>false</bool>
            </property>
            <property name="renameEnabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="targetModelLabel">
            <property name="text">
             <string>Target Model:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="qMRMLNodeComboBox" name="targetModelSelector">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="toolTip">
             <string>Select target model</string>
            </property>
            <property name="nodeTypes">
             <stringlist>
              <string>vtkMRMLModelNode</string>
             </stringlist>
            </property>
            <property name="noneEnabled">
             <bool>true</bool>
            </property>
            <property name="addEnabled">
             <bool>false</bool>
            </property>
            <property name="removeEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="scalingLabel">
            <property name="text">
             <string>Apply scaling</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="scalingCheckBox">
            <property name="toolTip">
             <string>Whether to apply scaling during alignment (highly recommended).</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="outputModelOptionalLabel">
            <property name="text">
             <string>Output registered model: </string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="qMRMLNodeComboBox" name="outputSelector">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="nodeTypes">
             <stringlist>
              <string>vtkMRMLModelNode</string>
             </stringlist>
            </property>
            <property name="noneEnabled">
             <bool>true</bool>
            </property>
            <property name="addEnabled">
             <bool>true</bool>
            </property>
            <property name="removeEnabled">
             <bool>true</bool>
            </property>
            <property name="renameEnabled">
             <bool>true</bool>
            </property>
            <property name="noneDisplay">
             <string>(Create new model)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="ctkCollapsibleButton" name="testPTDensityWidget">
         <property name="text">
          <string>Rigid registration</string>
         </property>
         <property name="collapsed">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="4" column="0" colspan="2">
           <widget class="QPlainTextEdit" name="subsampleInfo">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="plainText">
             <string/>
            </property>
            <property name="placeholderText">
             <string>Subsampling information</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="ctkSliderWidget" name="pointDensitySlider">
            <property name="toolTip">
             <string>Adjust the density of the pointclouds. Larger values increase the number of points, and vice versa.</string>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>2.000000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QPushButton" name="subsampleButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Run subsampling of the source and target pointclouds</string>
            </property>
            <property name="text">
             <string>Test pointcloud subsampling</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="pointDensityAdjustmentLabel">
            <property name="text">
             <string>Point Density Adjustment</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QPushButton" name="runRigidRegistrationButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Run a full pairwise ALPACA</string>
            </property>
            <property name="text">
             <string>Run rigid registration</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="ctkCollapsibleButton" name="runSingleRegistrationWidget">
         <property name="text">
          <string>Affine registration</string>
         </property>
         <property name="collapsed">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="runSingleALPACAWidgetLayout">
          <item row="2" column="0" colspan="2">
           <widget class="QPushButton" name="runCPDAffineButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Switch the the Advanced Settings tab to change parameters and run/re-run ALPACA</string>
            </property>
            <property name="text">
             <string>Run affine registration</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="advancedSettingsTab">
      <attribute name="title">
       <string>Advanced Settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="advancedSettingsLayout">
       <item>
        <widget class="ctkCollapsibleButton" name="pointDensityWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Point density and max projection</string>
         </property>
         <property name="collapsed">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="pointDensityWidgetLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="pointDensityAdvancedLabel">
            <property name="text">
             <string>Point Density Adjustment:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="ctkSliderWidget" name="pointDensityAdvancedSlider">
            <property name="toolTip">
             <string>Adjust the density of the pointclouds. Larger values increase the number of points, and vice versa.</string>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>3.000000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="ctkCollapsibleButton" name="rigidRegistrationWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Rigid registration</string>
         </property>
         <property name="collapsed">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="rigidRegistrationWidget">
          <item row="0" column="0">
           <widget class="QLabel" name="normalSearchRadiusLabel">
            <property name="text">
             <string>Normal Search Radius:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="ctkSliderWidget" name="normalSearchRadiusSlider">
            <property name="toolTip">
             <string>Set size of the neighborhood radius used when computing normals</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>10.000000000000000</double>
            </property>
            <property name="value">
             <double>2.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="FPFHSearchRadiusLabel">
            <property name="text">
             <string>FPFH Search radius:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="ctkSliderWidget" name="FPFHSearchRadiusSlider">
            <property name="toolTip">
             <string>Set size of the neighborhood used when computing FPFH features</string>
            </property>
            <property name="minimum">
             <double>3.000000000000000</double>
            </property>
            <property name="maximum">
             <double>20.000000000000000</double>
            </property>
            <property name="value">
             <double>5.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="maximumCPDThresholdLabel">
            <property name="text">
             <string>Maximum corresponding point distance:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="ctkSliderWidget" name="maximumCPDThreshold">
            <property name="singleStep">
             <double>0.250000000000000</double>
            </property>
            <property name="minimum">
             <double>0.500000000000000</double>
            </property>
            <property name="maximum">
             <double>6.000000000000000</double>
            </property>
            <property name="value">
             <double>3.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="maxRANSACLabel">
            <property name="text">
             <string>Maximum RANSAC iterations:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="ctkDoubleSpinBox" name="maxRANSAC">
            <property name="toolTip">
             <string>Maximum number of iterations of the RANSAC algorithm</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="maximum">
             <double>10000000.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>1000000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="PointSubsamplelabel">
            <property name="text">
             <string>Poisson Point Subsample</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QCheckBox" name="poissonSubsampleCheckBox">
            <property name="toolTip">
             <string>If checked, ALPACA will use Poisson Point Sub-Sampling Method.</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="ICPDistanceThresholdLabel">
            <property name="text">
             <string>Maximum ICP distance</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="ctkSliderWidget" name="ICPDistanceThresholdSlider">
            <property name="toolTip">
             <string>Maximum ICP points-pair distance threshold</string>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>3.000000000000000</double>
            </property>
            <property name="value">
             <double>1.500000000000000</double>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="FPFHNeighborsLabel">
            <property name="text">
             <string>FPFH Neighbor Count:</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="ctkSliderWidget" name="FPFHNeighborsSlider">
            <property name="toolTip">
             <string>Set size of the max neighborhood points used when computing FPFH</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>10.000000000000000</double>
            </property>
            <property name="maximum">
             <double>200.000000000000000</double>
            </property>
            <property name="value">
             <double>100.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkDoubleSpinBox</class>
   <extends>QWidget</extends>
   <header>ctkDoubleSpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
